diff/autofd/autofd_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package autofd_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"gonum.org/v1/gonum/diff/autofd"
+)
+
+func TestDerivative(t *testing.T) {
+	for _, tc := range []struct {
```

Comment:
Please pull the table out of `TestDerivative` as `var derivativeTests` and s/tc/test/g.